firmware/src/midi/midi_sync.hh

```diff
@@ -0,0 +1,154 @@
+#pragma once
+#include "midi_message.hh"
+#include "midi_queue.hh"
+#include "midi_router.hh"
+#include <cstdint>
+#include <algorithm>
+#include <array>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+
+namespace MetaModule
+{
+
+class MidiSync {
+private:
+    MidiQueue midi_out_queue;
+    bool queue_subscribed = false;
+    
+    // Store last sent values by channel and CC number
+    // Key: pair of (midi_chan, cc_num), Value: last sent CC value
+    using SyncKey = std::pair<uint8_t, uint8_t>;
```

Comment:
Ah... I didn't think about MIDI channel when I recommended a using a fixed-size array of notes.
E.g. if a user had MIDI CC74 on Channels 1, 2, and 3 then only one of them would work.
Hmm... I guess we need to decide if we want 16x the amount of data or else use a custom data type for this, maybe based on this: https://github.com/4ms/cpputil/blob/main/util/sequential_map.hh
I can try and check performance